### ISSUE TITLE
icon missing  on Xfce

### ISSUE THREAD

**Issue by shmu26** [rafaelmardojai/blanket#84]
The icon for the active window on the Xfce panel is just a generic square box. 
However, the icon pinned to the panel (which merely serves as a shortcut) displays properly.
In the screenshot of the Xfce panel, the active icon for Blanket is the second from the top. 
Lower down, you can see how the pinned icon displays properly. It is under the Google Chrome icon. 

I am running the Flatpak of Blanket. Version 0.3.1

system info: 
```
[code]
System:    Host: <filter> Kernel: 5.8.0-3-amd64 x86_64 bits: 64 compiler: N/A 
           parameters: BOOT_IMAGE=/boot/vmlinuz-5.8.0-3-amd64 
           root=UUID=<filter> ro quiet splash 
           **Desktop: Xfce 4.14.2 tk: Gtk 3.24.5 info: xfce4-panel wm: xfwm4 dm: LightDM 1.26.0** 
           **Distro: MX-19.3_ahs_x64 patito feo May 31  2020 base: Debian GNU/Linux 10 (buster)** 
Machine:   Type: Desktop Mobo: ASUSTeK model: PRIME B365M-A v: Rev X.0x serial: <filter> 
           UEFI: American Megatrends v: 1203 date: 10/10/2019 
CPU:       Topology: 8-Core model: Intel Core i7-9700 bits: 64 type: MCP arch: Kaby Lake 
           family: 6 model-id: 9E (158) stepping: D (13) microcode: D6 L2 cache: 12.0 MiB 
           flags: avx avx2 lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 48000 
           Speed: 1786 MHz min/max: 800/4700 MHz Core speeds (MHz): 1: 1997 2: 1988 3: 1974 
           4: 1980 5: 1998 6: 1997 7: 1995 8: 2000 
           Vulnerabilities: Type: itlb_multihit status: KVM: Split huge pages 
           Type: l1tf status: Not affected 
           Type: mds status: Not affected 
           Type: meltdown status: Not affected 
           Type: spec_store_bypass 
           mitigation: Speculative Store Bypass disabled via prctl and seccomp 
           Type: spectre_v1 mitigation: usercopy/swapgs barriers and __user pointer sanitization 
           Type: spectre_v2 mitigation: Enhanced IBRS, IBPB: conditional, RSB filling 
           Type: srbds mitigation: TSX disabled 
           Type: tsx_async_abort mitigation: TSX disabled 
Graphics:  Device-1: Intel vendor: ASUSTeK driver: i915 v: kernel bus ID: 00:02.0 
           chip ID: 8086:3e98 
           Display: x11 server: X.Org 1.20.9 driver: modesetting unloaded: fbdev,vesa 
           resolution: 1920x1080~60Hz 
           OpenGL: renderer: Mesa Intel UHD Graphics 630 (CFL GT2) v: 4.6 Mesa 20.1.8 
           direct render: Yes 
Audio:     Device-1: Intel 200 Series PCH HD Audio vendor: ASUSTeK driver: snd_hda_intel 
           v: kernel bus ID: 00:1f.3 chip ID: 8086:a2f0 
           Device-2: JMTek LLC. type: USB driver: hid-generic,snd-usb-audio,usbhid 
           bus ID: 1-1.3:4 chip ID: 0c76:161f 
           Sound Server: ALSA v: k5.8.0-3-amd64 
Network:   Device-1: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet vendor: ASUSTeK 
           driver: r8169 v: kernel port: e000 bus ID: 03:00.0 chip ID: 10ec:8168 
           IF: eth0 state: up speed: 1000 Mbps duplex: full mac: <filter> 
Drives:    Local Storage: total: 2.26 TiB used: 716.40 GiB (31.0%) 
           ID-1: /dev/sda vendor: Kingston model: SA400S37480G size: 447.13 GiB block size: 
           physical: 512 B logical: 512 B speed: 6.0 Gb/s serial: <filter> rev: K1B3 scheme: GPT 
           ID-2: /dev/sdb vendor: Toshiba model: HDWD120 size: 1.82 TiB block size: 
           physical: 4096 B logical: 512 B speed: 6.0 Gb/s rotation: 7200 rpm serial: <filter> 
           rev: ACF0 scheme: GPT 
Partition: ID-1: / raw size: 100.59 GiB size: 98.76 GiB (98.18%) used: 25.53 GiB (25.9%) 
           fs: ext4 dev: /dev/sda5 
Sensors:   System Temperatures: cpu: 29.8 C mobo: 27.8 C 
           Fan Speeds (RPM): N/A 
Repos:     No active apt repos in: /etc/apt/sources.list 
           Active apt repos in: /etc/apt/sources.list.d/brave-browser-release.list 
           1: deb [arch=amd64] https://brave-browser-apt-release.s3.brave.com/ buster main
           Active apt repos in: /etc/apt/sources.list.d/debian-stable-updates.list 
           1: deb http://deb.debian.org/debian buster-updates main contrib non-free
           Active apt repos in: /etc/apt/sources.list.d/debian.list 
           1: deb http://deb.debian.org/debian buster main contrib non-free
           2: deb http://deb.debian.org/debian-security buster/updates main contrib non-free
           Active apt repos in: /etc/apt/sources.list.d/google-chrome.list 
           1: deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main
           Active apt repos in: /etc/apt/sources.list.d/microsoft-edge-dev.list 
           1: deb [arch=amd64] http://packages.microsoft.com/repos/edge/ stable main
           Active apt repos in: /etc/apt/sources.list.d/mx.list 
           1: deb http://ftp.linux.org.tr/mx/repo/mx/repo/ buster main non-free
           2: deb http://ftp.linux.org.tr/mx/repo/mx/repo/ buster ahs
           No active apt repos in: /etc/apt/sources.list.d/opera-stable.list 
           Active apt repos in: /etc/apt/sources.list.d/sidekick-browser.list 
           1: deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main
           Active apt repos in: /etc/apt/sources.list.d/skype-stable.list 
           1: deb [arch=amd64] https://repo.skype.com/deb stable main
           No active apt repos in: /etc/apt/sources.list.d/various.list 
Info:      Processes: 265 Uptime: 3h 56m Memory: 15.50 GiB used: 9.57 GiB (61.7%) Init: SysVinit 
           v: 2.93 runlevel: 5 default: 5 Compilers: gcc: 8.3.0 alt: 8 Shell: quick-system-in 
           running in: quick-system-in inxi: 3.0.36 
[/code]

```

![image](https://user-images.githubusercontent.com/13636596/103477249-a52cba00-4dc5-11eb-8d74-2468d00650f7.png)

**Comment by apandada1** [comment]
Is the same issue also present in .deb/ppa?

**Comment by shmu26** [comment]
Tell me where to get a .deb and I will give it a try. 
I don't use Ubuntu PPAs on MX Linux, it's liable to break the system.

**Comment by apandada1** [comment]
Here is the link to the .deb file. https://launchpad.net/~apandada1/+archive/ubuntu/blanket/+files/blanket_0.3.1-1~ubuntu20.04.1_all.deb

Here are the links for libhandy-1, which is a dependency, and may not be available in your system.
https://launchpad.net/~apandada1/+archive/ubuntu/blanket/+files/libhandy-1-0_0.90.0-0~202009060346~ubuntu20.04.1_amd64.deb
https://launchpad.net/~apandada1/+archive/ubuntu/blanket/+files/gir1.2-handy-1_0.90.0-0~202009060346~ubuntu20.04.1_amd64.deb
https://launchpad.net/~apandada1/+archive/ubuntu/blanket/+files/handy-1-examples_0.90.0-0~202009060346~ubuntu20.04.1_amd64.deb
https://launchpad.net/~apandada1/+archive/ubuntu/blanket/+files/libhandy-1-dev_0.90.0-0~202009060346~ubuntu20.04.1_amd64.deb

You can grab the .deb file from any PPA, e.g. this. https://launchpad.net/~apandada1/+archive/ubuntu/blanket/+packages (scroll down and click on the package name, which I did to get the links for .deb packages.)

**Comment by shmu26** [comment]
Yup, same issue with the .deb

**Comment by johnfactotum** [comment]
I think this is a duplicate of #54.

**Comment by rafaelmardojai** [comment]
Yeah, looks like the same issue. @shmu26 could you test the [flatpak artifact from the last commit](https://github.com/rafaelmardojai/blanket/suites/1780687505/artifacts/34031437)?

**Comment by apandada1** [comment]
This seems to have been fixed in latest code in git with XFCE 4.16 (Manjaro).
Default Window Buttons plugin in XFCE. In my opinion, the default blanket icon would look better than this icon.
![Screenshot_2021-03-02_21-12-22](https://user-images.githubusercontent.com/3063132/110071178-9a7caa00-7da1-11eb-923d-1c733d026043.jpg)



With [xfce-doclike-plugin](https://github.com/nsz32/docklike-plugin) from AUR.
![Screenshot_2021-03-02_21-27-02](https://user-images.githubusercontent.com/3063132/110071200-a10b2180-7da1-11eb-84ec-b6915b56a44f.jpg)

**Comment by rafaelmardojai** [comment]
Can someone confirm if it's fixed in `0.4.0`?

**Comment by apandada1** [comment]
Yes, fixed.
![Screenshot_2021-04-16_11-23-36](https://user-images.githubusercontent.com/3063132/114978175-7fc04980-9ea6-11eb-922d-9103ceab3fc8.png)
And it also works with the xfce-docklike-plugin.
